pkg/model/estimator/local/ratio_process.go

```diff
@@ -42,7 +42,10 @@ func UpdateProcessComponentEnergyByRatioPowerModel(processMetrics map[uint64]*co
 
 	// if usageMetric exist, divide the power using the ratio. Otherwise, evenly divide the power.
 	if usageMetric != "" {
```

Comment:
Is cgroup metrics enabled?

I think if the hardware counters are disabled, we should use cpu time from cgroup metric instead.

Reply:
ok, add a check if cgroup is enabled, use it , I Think we have cgroup cpu time some where already
but
- how to calcualte the total usage based on cpu time, refer to estimate method?
- if no cgroup, no HW counter, we use -1 to avoid nil?

Reply:
I think that we should update the usage variables when HCs are disables during the Kepler initialization.

pkg/config/config.go
```
CoreUsageMetric       = getConfig("CORE_USAGE_METRIC", CPUInstruction)
DRAMUsageMetric       = getConfig("DRAM_USAGE_METRIC", CacheMiss)
UncoreUsageMetric     = getConfig("UNCORE_USAGE_METRIC", defaultMetricValue) // no metric (evenly divided)
GeneralUsageMetric    = getConfig("GENERAL_USAGE_METRIC", CPUInstruction)    // for uncategorized energy; pkg - core - uncore
```

The verification should be triggered in cmd/exporter.go

However, the cgroup metrics are not in the `CounterStats`, so we will still need to verify if the HW is not available and use the `CgroupStatMap` for containers and `SoftIRQCount` for processes.

Reply:
ok, let me update the whole logic

Reply:
ok, something like following ? 

```
if process

if HW COUNTER = true:
      systemcontainerMetrics.CounterStats[usageMetric]   ==> usageMetric = HW related, e.g cpu_cycles,  cpu_instr
else if CGROPU = true
      systemcontainerMetrics.Statmap[usageMetric] ==> usageMetric = cgroup related, e.g SOFTIRQ COUNT
else:
      report error
```

Reply:
if container:

if HW COUNTER = true:
      containerMetrics.CounterStats[usageMetric]   ==> usageMetric = HW related, e.g cpu_cycles,  cpu_instr
else if CGROPU = true:
      containerMetrics.CgroupStatMap[config.CgroupfsCPU] ==> use config.CgroupfsCPU
else 
      do nothing and log

Reply:
@marceloamaral why `CPUTime` not used for process ? why use SOFTIRQCOUNT ?

Reply:
> why CPUTime not used for process ? why use SOFTIRQCOUNT ?

Oh right! Let's use `CPUTime`. It will be a better metric.